runtime/resolver.js

```diff
@@ -41,11 +41,19 @@ class Resolver {
 
     var resolve; var reject;
     context.afterResolution = [];
+
     for (var component of recipe.components) {
       if (!new Resolver()._resolveComponent(context, component)) {
         return false;
       }
     }
+
+    let availableSlotIds = new Set([...arc.availableSlotIds || new Set()].concat(
```

Comment:
new Set() was because of the lazy initialization, fixed now.
in general, i was originally hoping that helper methods getRenderedSlotIds and getExposedSlotIds will make code better, but they did the opposite. So refactored everything.